moleculemaker/chemscraper-frontend
# Deploy production instance + set config at runtime

## Problem
We need a production instance of the new alphasynthesis tools: ChemScraper + MOLLI

## Approach
* Add production Helm values
* Change to use new namespace: `alphasynthesis`
* Added coming soon banner for consistency between ChemScraper + MOLLI + CLEAN
* Added login + logout for consistency with other alphasynthesis products
* Include configmap in Helm chart templates: `.Values.config`
    * Read configmap into environment variables in `templates/deployment.yaml`
    * Template environment variables into JSON file in entrypoint.sh: `src/assets/config/envvars.tpl.json` -> `src/assets/config/envvars.json`
    * At runtime, fetch envvars.json from the server via HTTP GET

## How to Test
Currently deployed in [production](https://chemscraper.frontend.mmli1.ncsa.illinois.edu)

1. Navigate to https://chemscraper.frontend.mmli1.ncsa.illinois.edu
2. Run a test job, and check the logs for `chemscraper-services` in the `alphasynthesis` (prod) namespace
    * You should see log messages appearing here, indicating the prod mmli-backend is talking to the prod chemscraper-services
4. Wait for the job to finish
    * You should see that the job completes successfully
